var/spack/repos/builtin/packages/openmx/package.py

```diff
@@ -83,10 +105,18 @@ def common_arguments(self):
 
     @property
     def build_targets(self):
-        return [
-            'openmx', 'DosMain', 'jx', 'analysis_example', 'esp', 'polB',
-            'bandgnu13', 'bin2txt', 'cube2xsf', 'intensity_map', 'md2axsf'
-        ] + self.common_arguments
+        if self.spec.satisfies('@3.9'):
+            return [
+                'openmx', 'DosMain', 'jx', 'analysis_example', 'esp', 'polB',
+                'calB', 'Z2FH', 'bandgnu13', 'bin2txt', 'cube2xsf', 'intensity_map',
+                'md2axsf', 'tp', 'kSpin', 'BandDispersion', 'ADenBand', 'FermiLoop',
+                'GridCalc', 'MulPOnly', 'MulPCalc', 'example_mpi_spawn', 'gcube2oned'
+            ] + self.common_arguments
+        else:
+            return [
+                'openmx', 'DosMain', 'jx', 'analysis_example', 'esp', 'polB',
+                'bandgnu13', 'bin2txt', 'cube2xsf', 'intensity_map', 'md2axsf'
+            ] + self.common_arguments
```

Comment:
This is a subset of the case above, so we can start with assigning it and adding more targets if the condition is met.

Reply:
Really don't know why `all` target is not used in the first place

Reply:
Needs to append `common_arguments()` call to overwrite compilers and libs vars. Testing now with 
```
    @property
    def build_targets(self):
        return ['all'] + self.common_arguments

```